BUILDING.md

```diff
@@ -29,11 +29,86 @@ This section contains package maintenance procedures in preparation for deployme
   git push --tags
   ```
 * Create a new release from the new tag on GitHub and let GitHub auto-generate release notes. Mark releases used for testing as pre-release. The release bundles the code into an archive, which we'll use for deployment.
+* Pushing a new tag will generate a new Docker image with the same tag.
 
 # Deployment
 This section contains instructions to deploy Build and the related service build2xlsform to a test or production server.
 
-## Deploy build2xlsform
+## docker-compose
+This section explains how to run ODK Build with docker-compose.
+This is useful for several audiences:
+
+* End users can run this as an offline or self-hosted version of ODK Build. Note, the installation process requires internet connectivity.
+* Developers can verify that the Docker builds still work after code changes.
+* Maintainers can deploy Build with the same toolset (docker-compose) as Central.
+
+### First time start-up
+With Docker and docker-compose installed, run
+
+```
+# via HTTPS
+git clone https://github.com/getodk/build.git 
+
+# or with SSH keys
+git clone git@github.com:getodk/build.git
+
+# or download and extract the latest release from https://github.com/getodk/build/releases
+# Replace 0.4.0 with your release version
+export BV="0.4.0"
+# Verify that the correct release version will be used
+echo $BV
+
+wget https://github.com/getodk/build/archive/$BV.tar.gz
+tar -xf $BV.tar.gz && mv build-$BV build && rm $BV.tar.gz
+```
+
+#### Current ODK Build master
+The file `docker-compose.dev.yml` will deploy the latest tagged image for build2xlsform from ghcr.io,
+and build the current master as ODK Build image.
+Developers will use this option to test ODK Build during development.
+
+```
+cd build
+docker-compose -f docker-compose.dev.yml up -d --build
+```
+
+This will run ODK Build on `http://0.0.0.0:9393/`, together with its Postgres database 
+and the related service `build2xlsform` providing an export to XLSForm.
+
+#### Latest official images
+The file `docker-compose.yml` will deploy the latest tagged images from ghcr.io,
+which are built and pushed whenever a new Git tag is pushed to GitHub.
+Maintainers will use this option to deploy ODK Build to a staging or production server.
+
+```
+docker-compose up -d
+```
+
+### Stop Build
+To stop the images, run 
+
+```
+docker-compose stop
+```
+
+DO NOT run `docker-compose d*wn` as it unlinks the database volume.
```

Comment:
why the `*`? in case someone copy pastes?

Reply:
also, we should just name the volume this time which we failed to understand to do for central. then down can be run safely as the named volume will persist.

Reply:
> why the `*`? in case someone copy pastes?

[Correct - remove beans where noses are present](https://articles.uie.com/beans-and-noses/)